### PR TITLE
[Kotlin Migration] CustomSyncServerUrlException 

### PR DESCRIPTION
**Pull request by codingtosh**
[Kotlin Migration] CustomSyncServerUrlException 
package com.ichi2.libanki.sync

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/sync/CustomSyncServerUrlException.kt [hunk 1]
```diff
@@ -14,33 +14,19 @@
...
-
-
-    private static String getMessage(String url) {
-        return "Invalid Custom Sync Server URL: " + url;
-    }
-
-
-    @Override
-    public String getLocalizedMessage() {
+class CustomSyncServerUrlException(val url: String, ex: IllegalArgumentException?) : RuntimeException(getMessage(url), ex) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/libanki/sync/CustomSyncServerUrlException.kt:21]: `cleanup: see if ex can be made non-null`

**Review by david-allison-1 — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/libanki/sync/CustomSyncServerUrlException.kt (hunk 1 above)
Comment (codingtosh) [reply AnkiDroid/src/main/java/com/ichi2/libanki/sync/CustomSyncServerUrlException.kt]: Done, PTAL.

**Review by mikehardy — APPROVED** [review]
comments resolved, LGTM, checking locally after rebase to master, if it passes I'll merge

**Comment by mikehardy** [comment]
Hi there! MARCH OpenCollective Notice

Just a friendly notice that we try to process OpenCollective payments monthly - it's time for [March 2022 submissions](https://github.com/ankidroid/Anki-Android/pulls?q=is%3Apr+merged%3A2022-03-01..2022-03-31)

If you are interested in compensation for this work, the process with details is here:

https://github.com/ankidroid/Anki-Android/wiki/OpenCollective-Payment-Process#how-to-get-paid

I only post one comment per person to avoid spamming you, regardless of the number of PRs merged, but this note applies to all PRs merged for in the month of March.

Please note that GSoC contributions are okay for this process. Our philosophy is that our users have donated to AnkiDroid for *all* contributions. The only PRs that will not go through the OpenCollective process are ones *directly* related to a GSoC project from a selected participant, since those receive a stipend from GSoC itself.

Please understand that this means the monthly budget we have for payments might be stretched a bit and the cap on payments-per-person may be lower, but we try to make our process as fair and transparent as possible, we just need your understanding.

Thanks!

**Pull request merged**